pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,129 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/dcos/dcos-cli/pkg/setup"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	setupFlags := setup.NewFlags(ctx.Fs(), ctx.EnvLookup)
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			manager := ctx.ConfigManager()
+			linkableClusterConfig, err := manager.Find(args[0], false)
+			if err != nil {
+				return err
+			}
+
+			if config.Empty() == linkableClusterConfig {
+				// The cluster does not exist yet. Two possibilities:
+				// - The argument is an URL, we try to setup the cluster.
+				// - The argument is not an URL, we return an error.
+				_, err = url.ParseRequestURI(args[0])
+				if err != nil {
+					return errors.New("unable to retrieve cluster " + args[0])
+				}
+
+				msg := " is not set up in the CLI, would you like to do it now?"
+				err = ctx.Prompt().Confirm(args[0] + msg)
+				if err != nil {
+					return err
+				}
+
+				// We do not want to attach the linkable cluster.
+				attach := false
+				ctx.Setup(setupFlags, args[0], attach)
+			}
+
+			if attachedCluster.Config() == linkableClusterConfig {
```

Comment:
This wouldn't compare equality as these are pointers. You could either compare their paths, or even skip this check and let the API return an error.